scheduler-server/scheduling-dsl-compiler/src/libs/scheduler-edsl-fluent-api.ts

```diff
@@ -162,6 +170,8 @@ declare global {
 
     public or(...others: Goal[]): Goal
 
+    public static applyWhen(opts: { goal: Goal, window: WindowsEDSL.Windows } ): Goal
```

Comment:
Have you tested this directly in the UI ?

Reply:
Yes, I was able to recreate the test from SchedulingIntegrationTests in the UI and it worked!
<img width="1158" alt="image" src="https://user-images.githubusercontent.com/50245172/176248115-ce5e155c-d440-4e4b-a746-4d5f18be872a.png">